internal/host/validator.go

```diff
@@ -305,6 +304,81 @@ func (v *validator) isMediaConnected(c *validationContext) (ValidationStatus, st
 	}
 }
 
+// isMtuOK - This validation distinguishes between CMN and UMN. For UMN, the MTU is checked across all interfaces, while for CMN, it focuses specifically on the MTU reports for the machine networks.
+func (v *validator) isMtuOK(c *validationContext) (ValidationStatus, string) {
+	if c.infraEnv != nil {
+		return ValidationSuccessSuppressOutput, LateBindingMsg
+	}
+	if c.inventory == nil {
+		return ValidationPending, "Missing host inventory"
+	}
+	if common.IsSingleNodeCluster(c.cluster) {
+		return ValidationSuccessSuppressOutput, "The validation is not applicable to SNO"
+	}
+	if len(c.host.Connectivity) == 0 {
+		return ValidationPending, "Missing MTU report information"
+	}
+	connectivityReport, err := hostutil.UnmarshalConnectivityReport(c.host.Connectivity)
+	if err != nil {
+		return ValidationError, "Internal error - failed to parse host connectivity report"
+	}
+
+	if swag.BoolValue(c.cluster.UserManagedNetworking) {
+		return v.isMtuOKAllInterfaces(c, connectivityReport)
+	}
+	return v.isMtuOKInMachineNetwork(c, connectivityReport)
+}
+
+// isMtuOKAllInterfaces - intended for UMN, where the MTU report is checked across all interfaces.
+func (v *validator) isMtuOKAllInterfaces(c *validationContext, connectivityReport *models.ConnectivityReport) (ValidationStatus, string) {
+	for _, r := range connectivityReport.RemoteHosts {
+		for _, mtuReport := range r.MtuReport {
+			if !mtuReport.MtuSuccessful {
+				return ValidationFailure, fmt.Sprintf("MTU is broken. Interface %s, remote IP address: %s", mtuReport.OutgoingNic, mtuReport.RemoteIPAddress)
+			}
+		}
+	}
+	return ValidationSuccess, "MTU is ok"
+}
+
+// isMtuOKInMachineNetwork - intended for CMN, focusing exclusively on the MTU reports for the machine network.
+func (v *validator) isMtuOKInMachineNetwork(c *validationContext, connectivityReport *models.ConnectivityReport) (ValidationStatus, string) {
+	for _, machineNet := range c.cluster.MachineNetworks {
+		_, mNetwork, err := net.ParseCIDR(string(machineNet.Cidr))
+		if err != nil {
+			return ValidationError, "Internal error - failed to parse machine network CIDR"
+		}
+		for _, r := range connectivityReport.RemoteHosts {
+			for _, mtuReport := range r.MtuReport {
+				var sourceIP string
+				for _, nic := range c.inventory.Interfaces {
```

Comment:
Maybe put the test against the inventory in a function ?
Can you first check if the remoteIP is in the machineNetwork and then the local IP ?

Reply:
> Maybe put the test against the inventory in a function ?
Yes, I agree

> Can you first check if the remoteIP is in the machineNetwork and then the local IP ?
Can you explain why it is important?

Reply:
> Can you explain why it is important?

Actually it is OK if you leave the order as is